docs/cloud/rahti/tutorials/connect-database-hpc.md

```diff
@@ -88,7 +90,7 @@ export PATH=$PATH:$PWD
 websocat -b tcp-l:127.0.0.1:0 wss://websocat-<project name>.rahtiapp.fi -E &
 ws_pid=$!  # $! contains the process ID of the most recently executed background command
 mkdir -p /tmp/$USER
-lsof -i -p $ws_pid 2>/dev/null | grep TCP | grep -oE "localhost:[0-9]*" | cut -d ":" -f2 > /tmp/$USER/${SLURM_JOB_ID}_rahtidb_port
+lsof -Pa -p $ws_pid -i -sTCP:LISTEN | grep -v ^COMMAND | sed -e 's#.*TCP\s*\S*:\([0-9]*\).*#\1#' > /tmp/$USER/${SLURM_JOB_ID}_rahtidb_port
```

Comment:
This command will only return the port of the most recently websocat process.
If you have another localhost process for other purposes (i.e python3 -m http.server --bind localhost),
this command won't return the python process.

Reply:
`-p $ws_pid` will filter by the PID in the varaible `$ws_pid`. At least this is the case in Linux

Reply:
```
# galvaro @ kaki in ~ [15:34:03] 
$ python -m http.server & ws_pid=$!
[1] 18701

# galvaro @ kaki in ~ [15:34:15] 
$ Serving HTTP on 0.0.0.0 port 8000 (http://0.0.0.0:8000/) ...
echo $ws_pid
18701
````

and then:

```
# galvaro @ kaki in ~ [15:36:54] 
$ lsof -Pa  -i -sTCP:LISTEN | grep -v ^COMMAND 
python  18701 galvaro    3u  IPv4 257478      0t0  TCP *:8000 (LISTEN)
python  18854 galvaro    3u  IPv4 259592      0t0  TCP *:27868 (LISTEN)
python  18965 galvaro    3u  IPv4 256444      0t0  TCP *:30903 (LISTEN)

# galvaro @ kaki in ~ [15:36:59] 
$ lsof -Pa -p $ws_pid -i -sTCP:LISTEN | grep -v ^COMMAND
python  18701 galvaro    3u  IPv4 257478      0t0  TCP *:8000 (LISTEN)
```

Reply:
Ok, we were saying the same. I misunderstood